### PR TITLE
feat(MenuItem): allow Items to override the closeOnSelect behaviour of a MenuTrigger

### PR DESCRIPTION
**Pull request by dominiccarrington**
# Description

This PR adds the closeOnSelect attribute to the react-aria Item component and allows an Item component to override the default behaviour defined by <MenuTrigger>

# Links

Required for: https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-505153

### CODE REVIEW

File: src/components/MenuItem/MenuItem.unit.test.tsx [hunk 1]
```diff
@@ -240,4 +243,13 @@ describe('<MenuItem />', () => {
 
     expect(onCloseMock).not.toHaveBeenCalled();
   });
+
+  it('should call useMenuContext with closeOnSelect override', async () => {
```
Comment (nataliadelmar) [inline src/components/MenuItem/MenuItem.unit.test.tsx:247]: We could have another test for 'should call useMenuContext with empty object {} for item key ="$.1"

File: src/components/MenuTrigger/MenuTrigger.stories.tsx [hunk 2]
```diff
@@ -214,4 +215,69 @@ Common.parameters = {
...
+        <Menu key="2">
+          <SelectionGroup key="test1" selectionMode="single" title="Will keep the menu open">
+            <Item key="one">Selection 1</Item>
+            <Item key="two">Selection 2</Item>
+            <Item key="three">Selection 3</Item>
+          </SelectionGroup>
+          <ContentSeparator />
+          <Section key="test2">
+            <Item key="close" closeOnSelect>
+              Action
```
Comment (nataliadelmar) [inline src/components/MenuTrigger/MenuTrigger.stories.tsx:246]: nice! "Action that will close the menu once triggered"

**Review by nataliadelmar — COMMENTED** [review]
lgtm!

**Review by nataliadelmar — APPROVED** [review]
